src/components/atoms/UiCheckbox/UiCheckbox.vue

```diff
@@ -69,82 +69,87 @@ import {
   computed,
   useSlots,
 } from 'vue';
-import type { PropType } from 'vue';
+import type {
+  LabelHTMLAttributes,
+  InputHTMLAttributes,
+} from 'vue';
 import equal from 'fast-deep-equal';
 import { uid } from 'uid/single';
-import type { PropsAttrs } from '../../../types/attrs';
-import type { Icon } from '../../../types/icon';
-import type { HTMLTag } from '../../../types/tag';
-import UiIcon from '../UiIcon/UiIcon.vue';
-import UiText from '../UiText/UiText.vue';
 import useAttributes from '../../../composable/useAttributes';
 import { keyboardFocus as vKeyboardFocus } from '../../../utilities/directives';
+import UiIcon from '../UiIcon/UiIcon.vue';
+import type { IconAttrsProps } from '../UiIcon/UiIcon.vue';
+import UiText from '../UiText/UiText.vue';
+import type { TextAttrsProps } from '../UiText/UiText.vue';
+import type { DefineAttrsProps } from '../../../types';
 
-export type CheckboxValue = string | Record<string, unknown>[];
-export type CheckboxModelValue = boolean | Record<string, unknown>[];
-const props = defineProps({
+export interface CheckboxProps {
   /**
    *  Use this props or v-model to set checked.
    */
-  modelValue: {
-    type: [
-      Boolean,
-      Array,
-    ] as PropType<CheckboxModelValue>,
-    default: false,
-  },
+  modelValue?: boolean | Record<string, unknown>[];
   /**
    * Use this props to set value of checkbox.
    * Required for multiple checkboxes.
    */
-  value: {
-    type: [
-      String,
-      Object,
-    ] as PropType<CheckboxValue>,
-    default: '',
-  },
+  value?: string | Record<string, unknown>;
```

Comment:
Q: Is value and modelValue related? I mean, is it connected in a way that if `modelValue` is a boolean, then `value` is a strong or if `modelValue` is a `Record`, then value is a `Record` too? 🤔 :)

Reply:
IMO yes, is related? @ArkadiuszKrzymienski can you check it?

Reply:
If it's related, then we should have an union of possible types, like:
```typescript
export type CheckboxProps = ({
    /**
     *  Use this props or v-model to set checked.
     */
    modelValue?: boolean
    /**
     * Use this props to set value of checkbox.
     * Required for multiple checkboxes.
     */
    value?: string
    /**
     * Use this props to set checkbox id.
     * Default value is generated by uid i.e. `checkbox-<uid>`.
     */
} | {
    /**
     *  Use this props or v-model to set checked.
     */
    modelValue?: Record<string, unknown>[];
    /**
     * Use this props to set value of checkbox.
     * Required for multiple checkboxes.
     */
    value?: Record<string, unknown>;
    /**
     * Use this props to set checkbox id.
     * Default value is generated by uid i.e. `checkbox-<uid>`.
     */
}) & {
    id?: string;
    /**
     * Use this props to disabled checkbox.
     * Remember to use `ui-checkbox--is-disabled` class to style disabled checkbox.
     */
    disabled?: boolean;
    /**
     * Use this props to pass attrs for input element.
     */
    inputAttrs?: DefineAttrsProps<null, InputHTMLAttributes>;
    /**
     * Use this props to pass attrs for checkmark UiIcon
     */
    iconCheckmarkAttrs?: IconAttrsProps;
    /**
     * Use this props to pass attrs for label UiText
     */
    textLabelAttrs?: TextAttrsProps;
}
```

Reply:
yeah, it's related. Great idea, thanks @AlanLes 👍

Reply:
@AlanLes Unfortunately, when I tried this solution, I got the error `argument passed to defineProps() must be a literal type, or a reference to an interface or literal type.` 😕 Am I doing something wrong or is it not possible?

Reply:
Ahhhh, I didn't know that about Vue.. :( Looks like it's not possible to correlate types of several props in an easy way. :/  Sorry

Reply:
It would be great feature though :D

Reply:
yeah it is a great idea 💪 I will follow this topic 👀